### PR TITLE
Newsletters: add 208 (2022-07-13)

### PR DESCRIPTION
**Pull request by harding**
- [x] Add lede, update releases/RCs, add topic links @harding
- [x] Add topic for LN probing @harding
- [x] Bitcoin Core PR Review Club summary @glozow 
- [x] Bitcoin Core 25353 Add a `-mempoolfullrbf` node setting @Xekyo 
- [x] Bitcoin Core 25454 p2p: Avoid multiple getheaders messages in flight to the same peer @adamjonas

### CODE REVIEW

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+- **X-only workaround:** Bitcoin public keys are points on
+  two-dimensional graph which are referenced by the intersection of an
+  *x* coordinate and a *y* coordinate.  For any given *x* coordinate,
+  there are only two valid *y* coordinates and these can be calculated
+  from the *x* value.  This allowed an optimization in the design of
+  [taproot][topic taproot] to require all public keys used with
+  [BIP340][]-style schnorr signatures to only use a certain one of these
+  *y* coordinates, allowing any public keys included in transactions to
+  omit including the *y* point entirely, saving one vbyte per signature
+  over the original taproot design..
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-07-13-newsletter.md:38]:
```suggestion
  over the original taproot design.
```

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    A the time, this technique (called *x-only public keys*) was thought
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-07-13-newsletter.md:40]:
```suggestion
    At the time, this technique (called *x-only public keys*) was thought
```

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    well.
+
+    This week, Tim Ruffing [posted][ruffing xonly] to the Bitcoin-Dev
+    mailing list a potential workaround that only requires a slight bit
+    of additional computation by signers---an amount that even a
+    resource-constrained hardware signing device can probably perform
+    without making its user wait too long.
+
+- **Allowing deliberately slow LN payment forwarding:** in a reply to a
+  thread about recursive/nested MuSig2 (see [Newsletter #204][news204
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-07-13-newsletter.md:55]:
```suggestion
  thread about recursive/nested [MuSig2][topic musig] (see [Newsletter #204][news204
```

**Comment by Xekyo** [comment]
Given that the PR I described this week has potential to rile up people, I would appreciate if reviewers especially read my description from a view point of a Bitcoin user that is averse to general full RBF. Please let me know if you see ways the description could be more defusive.

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  be used for safely reporting security vulnerabilities to the LDK
+  maintainers.
+
+- [BTCPay Server #3922][] adds the basic UI for *custodian
+  accounts*---accounts tied into a BTCPay instance where the funds are
+  managed by a custodian, such as a Bitcoin exchange (rather than by the
+  local user controlling their own private keys).  BTCPay instances may
+  have both local wallets and custodian accounts, which can make it
+  easy to manage funds between them, e.g. allowing a merchant to receive
+  funds privately and security to their wallet but also quickly transfer
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-07-13-newsletter.md:150]:
```suggestion
  funds privately and securely to their wallet but also quickly transfer
```

**Review by bitschmidty — COMMENTED** [review]
Looking good so far! A few suggestions

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #25353][] introduces the `mempoolfullrbf` configuration
+  option which may be used to generally permit transaction replacement.
+  Currently, Bitcoin Core will only allow replacement of transactions in
+  its mempool that have signaled replaceability as defined per
+  [BIP125][]. This behavior is referred to as "opt-in full RBF" which
+  continues to be the default behavior of nodes after this change. The
+  new `mempoolfullrbf` option permits a node operator to change their
+  node's replacement behavior to "general full RBF" which permits
```
Comment (glozow) [inline _posts/en/newsletters/2022-07-13-newsletter.md:110]: Question, would you be able to point to where {opt-in, general} {full, not full} RBF terms are defined?

Comment (Xekyo) [reply _posts/en/newsletters/2022-07-13-newsletter.md:151]: For opt-in RBF maybe:

- https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-November/011783.html
- https://bitcoincore.org/en/faq/optin_rbf/
- https://bitcoin.stackexchange.com/q/38320/5406

I'm not sure whether anyone else previously used "general full RBF". The alternative proposal was called "first-seen-safe RBF", which required that all the outputs of the replaced transactions had to be present in the replacement transaction with the same or a higher amount.

Comment (harding) [reply _posts/en/newsletters/2022-07-13-newsletter.md:151]: I think this summary might benefit by linking to when announce this was a possible upcoming change in https://bitcoinops.org/en/newsletters/2022/06/22/#full-replace-by-fee .  I think that could reduce the likeliness that people would get upset by your description---we can just point out that we warned people that the idea was on the table.  (We also warned them last time this came up too: https://bitcoinops.org/en/newsletters/2021/06/23/#allowing-transaction-replacement-by-default )

I think a link to our RBF topic page might also be helpful: https://bitcoinops.org/en/topics/replace-by-fee/

---

Regarding terminology, if anyone is feeling ambitious, maybe open a PR to our RBF topic page where you define the terms you think are relevant.  If you make it a list with bold terms (like in https://bitcoinops.org/en/topics/op_checksigfromstack/), you'll be able to link to the specific terms.

I don't care what term is used here but historically Peter Todd, who chose the terms, has used "opt-in full-RBF" for (what became) BIP125 and "full-RBF" for signaling-not-required replacements.  See, e.g., https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-January/013471.html

In Optech, I've preferred to use "opt-in RBF" for BIP125 and "full RBF" for non-signaled, e.g. https://bitcoinops.org/en/newsletters/2022/06/22/#full-replace-by-fee

Comment (glozow) [reply _posts/en/newsletters/2022-07-13-newsletter.md:151]: > Peter Todd, who chose the terms, has used "opt-in full-RBF" for (what became) BIP125 and "full-RBF" for signaling-not-required replacements. See, e.g., https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-January/013471.html

> In Optech, I've preferred to use "opt-in RBF" for BIP125 and "full RBF" for non-signaled, e.g. https://bitcoinops.org/en/newsletters/2022/06/22/#full-replace-by-fee

This is also how I use them, i.e. opt-in means you apply signaling rules, full means you don't. I don't think we should  use "general full rbf" if it's a new term.

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 6]
```diff
@@ -100,7 +100,16 @@ Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
 Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
 Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
 
-- [Bitcoin Core #25353][] Add a `-mempoolfullrbf` node setting FIXME:Xekyo
+- [Bitcoin Core #25353][] introduces the `mempoolfullrbf` configuration
```
[the remark this answers is not in the document]
Comment (Xekyo) [reply _posts/en/newsletters/2022-07-13-newsletter.md:144]: Thanks, I put this in when rewriting my description.

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 7]
```diff
@@ -71,13 +71,54 @@ This week's newsletter FIXME
...
+added. On the other hand, `addnode` mode causes the node to repeatedly
+try to connect to the node until it succeeds."
+  a1link="https://bitcoincore.reviews/24170#l-72"
+
+  q2="The PR introduces a new connection type `MANUAL_BLOCK_RELAY`
+that combines the properties of `MANUAL` and `BLOCK_RELAY` peers. What
+are the advantages and disadvantages of having an extra connection
+type, as opposed to combining the logic of the existing ones?"
+  a2="As there are many attributes of p2p connections but few types,
+using flat, using enumerated connection types is simpler. Some noted
```
[the remark this answers is not in the document]
Comment (glozow) [reply _posts/en/newsletters/2022-07-13-newsletter.md]: Woops, thanks!

**Review by Xekyo — COMMENTED** [review]
LGTM

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+participants agreed that using flat, enumerated connection types is
+simpler. They also noted that describing connections using
+combinations of capabilities and permissions could lead to a
+combinatorial blowup of connection types and convoluted logic,
+including some which might not make sense."
+  a2link="https://bitcoincore.reviews/24170#l-97"
+
+  q3="What types of attacks that this PR tries to mitigate are fixed
+by BIP324? Which ones aren’t?"
+  a3="BIP324 enhances privacy by adding opportunistic encryption to
```
Comment (bitschmidty) [inline _posts/en/newsletters/2022-07-13-newsletter.md:117]:
```suggestion
  a3="[BIP324][] enhances privacy by adding opportunistic encryption to
```

**Comment by harding** [comment]
Made edits (thanks for the feedback @Xekyo @glozow @bitschmidty !) and added remaining content.  Reviewed contributions (thanks @Xekyo @glozow @adamjonas !) and everything LGTM.  Thanks everyone!

File: _posts/en/newsletters/2022-07-13-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+  individual signatures.  Importantly, this scheme is non-interactive,
+  which means that a set of signatures can be half-aggregated by a third
+  party without any involvement from the signers."
+
+    A [separate document][cia doc] provides examples of how half
+    aggregation could benefit the operators of Bitcoin and LN nodes, plus
+    several concerns that would need to be considered in the design of a
+    soft fork adding half aggregation to the consensus protocol.
+
+- **X-only workaround:** Bitcoin public keys are points on
```
Comment (glozow) [inline _posts/en/newsletters/2022-07-13-newsletter.md:35]:
```suggestion
- **X-only workaround:** Bitcoin public keys are points on a
```
?

**Review by glozow — COMMENTED** [review]
LGTM

File: _posts/en/newsletters/2022-07-13-newsletter.md (hunk 5 above)
Comment (glozow) [reply _posts/en/newsletters/2022-07-13-newsletter.md]: Thank you for being open-minded! :) I learned something new about RBF history from this discussion. The summary now looks :ok_hand: to me.

**Comment by bitschmidty** [comment]
Took @glozow's suggestion (https://github.com/bitcoinops/bitcoinops.github.io/pull/808#discussion_r919802277), fixed (my own bad, sorry!) suggestion for link to BIP324 (not on bips repo), squashed, and merged.

Thank you team! @harding @Xekyo @glozow @adamjonas

**Pull request merged**